cli/cmd/initialize.go

```diff
@@ -41,39 +48,51 @@ func handleTLSOptions() {
 			},
 		}
 		_ = survey.AskOne(modePrompt, &tlsMode)
+	} else {
+		tlsMode = Import
+	}
 
-		if tlsMode == Generate {
-			// Generate mode requires a host entry
-			prompt := &survey.Input{
-				Message: "Enter a host DNS entry or IP Address for the cluster ingress",
-			}
-			_ = survey.AskOne(prompt, &initOptions.PKI.Host, survey.WithValidator(survey.Required))
-		} else {
-			// Import mode requires the public and private key paths
-			prompt := &survey.Input{
-				Message: "Enter a file path to the ingress public key",
-				Suggest: func(toComplete string) []string {
-					// Give some suggestions to users
-					files, _ := filepath.Glob(toComplete + "*")
-					return files
-				},
-			}
-			_ = survey.AskOne(prompt, &initOptions.PKI.CertPublicPath, survey.WithValidator(survey.Required))
+	// Always ask for a host entry to avoid having to guess which entry in a cert if provided
+	prompt := &survey.Input{
+		Message: "Enter a host DNS entry or IP Address for the cluster ingress. If using localhost, use 127.0.0.1",
+	}
+	_ = survey.AskOne(prompt, &state.TLS.Host, survey.WithValidator(survey.Required))
 
-			prompt.Message = "Enter a file path to the ingress private key"
-			_ = survey.AskOne(prompt, &initOptions.PKI.CertPrivatePath, survey.WithValidator(survey.Required))
+	if tlsMode != Generate {
+		// Import mode requires the public and private key paths
+		prompt := &survey.Input{
+			Message: "Enter a file path to the ingress public key",
+			Suggest: func(toComplete string) []string {
+				// Give some suggestions to users
+				files, _ := filepath.Glob(toComplete + "*")
+				return files
+			},
 		}
+		_ = survey.AskOne(prompt, &state.TLS.CertPublicPath, survey.WithValidator(survey.Required))
+
+		prompt.Message = "Enter a file path to the ingress private key"
+		_ = survey.AskOne(prompt, &state.TLS.CertPrivatePath, survey.WithValidator(survey.Required))
 	}
-	if !utils.CheckHostName(initOptions.PKI.Host) {
-		logrus.Fatalf("The hostname provided (%v) was not a valid hostname. The hostname can only contain: 'a-z', 'A-Z', '0-9', '-', and '.' characters.\n", initOptions.PKI.Host)
+
+	if !utils.CheckHostName(state.TLS.Host) {
+		// On error warn user and cycle the function
+		logrus.Warnf("The hostname provided (%v) was not a valid hostname. The hostname can only contain: 'a-z', 'A-Z', '0-9', '-', and '.' characters as defined by RFC-1035.  If using localhost, you must use the 127.0.0.1.\n", state.TLS.Host)
+		handleTLSOptions()
+	} else {
+		if err := config.WriteState(state); err != nil {
+			logrus.Debug(err)
+			logrus.Fatal("Unable to save the zarf state file.")
+		}
 	}
 }
 
 func init() {
+	state := config.GetState()
```

Comment:
I'm going to remove this line. We defined `state` up above around line 18. This line overrides that variable but only exists in the local scope so we don't have the provided host/tls-paths users provided.